x/migration/keeper/msg_server_morse_account_state_test.go

```diff
@@ -3,23 +3,96 @@ package keeper_test
 import (
 	"testing"
 
+	authtypes "github.com/cosmos/cosmos-sdk/x/auth/types"
+	govtypes "github.com/cosmos/cosmos-sdk/x/gov/types"
 	"github.com/stretchr/testify/require"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
 
+	"github.com/pokt-network/poktroll/testutil/events"
 	keepertest "github.com/pokt-network/poktroll/testutil/keeper"
-	"github.com/pokt-network/poktroll/testutil/sample"
+	"github.com/pokt-network/poktroll/testutil/testmigration"
 	"github.com/pokt-network/poktroll/x/migration/keeper"
-	"github.com/pokt-network/poktroll/x/migration/types"
+	migrationtypes "github.com/pokt-network/poktroll/x/migration/types"
 )
 
-func TestMorseAccountStateMsgServerCreate(t *testing.T) {
+func TestMorseAccountStateMsgServerCreate_Success(t *testing.T) {
 	k, ctx := keepertest.MigrationKeeper(t)
 	srv := keeper.NewMsgServerImpl(k)
-	authority := sample.AccAddress()
-	expected, err := types.NewMsgCreateMorseAccountState(authority, types.MorseAccountState{})
+
+	numAccounts := 10
+	_, accountState := testmigration.NewMorseStateExportAndAccountState(t, numAccounts)
+
+	// Assert that the MorseAccountState is not set initially.
+	_, isFound := k.GetMorseAccountState(ctx)
+	require.False(t, isFound)
+
+	msgCreateMorseAccountState, err := migrationtypes.NewMsgCreateMorseAccountState(
+		authtypes.NewModuleAddress(govtypes.ModuleName).String(),
+		*accountState,
+	)
 	require.NoError(t, err)
 
-	_, err = srv.CreateMorseAccountState(ctx, expected)
+	res, err := srv.CreateMorseAccountState(ctx, msgCreateMorseAccountState)
 	require.NoError(t, err)
-	_, found := k.GetMorseAccountState(ctx)
-	require.True(t, found)
+
+	expectedUploadMsg := &migrationtypes.MsgCreateMorseAccountState{
```

Comment:
Do you mind adding small comments to the grouped sections?

As a reader of the code (who doesn't just LGTM but also doesn't have time to review everything line by line right now), here's how I review tests:

1. Look for "grouped comments"
2. Get an idea of the flow
3. Understand what is being done and what may or may not be missing